src/main/java/frc/utils/auto/PathPlannerUtils.java

```diff
@@ -0,0 +1,97 @@
+package frc.utils.auto;
+
+import com.pathplanner.lib.auto.AutoBuilder;
+import com.pathplanner.lib.auto.NamedCommands;
+import com.pathplanner.lib.commands.PathfindingCommand;
+import com.pathplanner.lib.controllers.PPHolonomicDriveController;
+import com.pathplanner.lib.path.GoalEndState;
+import com.pathplanner.lib.path.PathConstraints;
+import com.pathplanner.lib.path.PathPlannerPath;
+import com.pathplanner.lib.pathfinding.LocalADStar;
+import com.pathplanner.lib.pathfinding.Pathfinder;
+import com.pathplanner.lib.pathfinding.Pathfinding;
+import com.pathplanner.lib.util.HolonomicPathFollowerConfig;
+import edu.wpi.first.math.Pair;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Subsystem;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.BooleanSupplier;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+public class PathPlannerUtils {
+
+	private static List<Pair<Translation2d, Translation2d>> dynamicObstacles = List.of();
+
+	public static void startPathfinder() {
+		setPathfinder(new LocalADStar());
+		scheduleWarmup();
+	}
+
+	public static void setPathfinder(Pathfinder pathfinder) {
+		Pathfinding.setPathfinder(pathfinder);
+	}
+
+	public static void scheduleWarmup() {
+		PathfindingCommand.warmupCommand().schedule();
+	}
+
+	public static void configPathPlanner(
+		Supplier<Pose2d> poseSupplier,
+		Consumer<Pose2d> resetPose,
+		Supplier<ChassisSpeeds> robotRelativeSpeedsSupplier,
+		Consumer<ChassisSpeeds> robotRelativeSpeedsSetter,
+		HolonomicPathFollowerConfig holonomicPathFollowerConfig,
+		BooleanSupplier shouldFlipPath,
+		Subsystem driveSubsystem
+	) {
+		AutoBuilder.configureHolonomic(
+			poseSupplier,
+			resetPose,
+			robotRelativeSpeedsSupplier,
+			robotRelativeSpeedsSetter,
+			holonomicPathFollowerConfig,
+			shouldFlipPath,
+			driveSubsystem
+		);
+	}
+
+	public static void registerCommand(String commandName, Command command) {
+		NamedCommands.registerCommand(commandName, command);
+	}
+
+	public static void setDynamicObstacles(List<Pair<Translation2d, Translation2d>> obstacles, Pose2d currentRobotPose) {
```

Comment:
Only if the last path went through an obstacle 
In remove you can "lie" because the path won't be recalculated, but in set and add you need the real pose

Reply:
No obstacles so nothing blocking the path so the path doesn't need to be recalculated 

That is the only use of the pose in the function 
![Screenshot_2024-09-05-16-28-00-490_com android chrome](https://github.com/user-attachments/assets/19cac4da-a181-41da-9a60-b2a8dee911fc)

Reply:
weird because if i am in the middle of a path and remove obstacles it "clear" my way and can do quicker path.

Reply:
Maybe there's a way to force recalculate, but then we definitely need the real pose

Reply:
i dont know if it can harm to not give pose, and i think we will use this func in functions that gets pose (for the path) so maybe its better to get the pose

Reply:
I'm sure the pathfinder set obstacles function won't set the current pose in remove, but maybe the pathfinding thread recalculates the path automatically. I need to read more to be sure

Reply:
Ok so the thread only recalculates the path if goal or starting pose is set. 
If you want to force recalculation I can set starting position in remove (it could make a faster path but it will increase code complexity)